### PR TITLE
build: add directory for node tests

### PR DESCRIPTION
**Pull request by justinj**
This is still just a translation of the existing test, but I think we
can do a lot to better structure the tests here in the future.

### CODE REVIEW

File: pkg/acceptance/node_test.go [hunk 1]
```diff
@@ -28,120 +28,14 @@ func TestDockerNodeJS(t *testing.T) {
...
-    client.end(function (err) {
-      if (err) throw err;
-    });
-  }).catch(err => {
-    client.end()
-    throw err;
-  });
-});
+export SHOULD_FAIL=%v
+npm install && npm test
```
Comment (tschottdorf) [inline pkg/acceptance/node_test.go]: Does this work offline?

Comment (justinj) [reply pkg/acceptance/node_test.go]: Yes, as long as it's been run before to install the dependencies once, it won't try to hit the network to pull them down again (not that I had realized this was a concern beforehand).

Comment (tschottdorf) [reply pkg/acceptance/node_test.go]: That's fine, was just checking.

**Review by tbg — APPROVED** [review]
LGTM, but if you wanted an expert review on the nodeJS you'll want to ask someone else. :)

Thanks for refactoring those, by the way!

File: pkg/acceptance/node_test.go (hunk 1 above)
Comment (justinj) [reply pkg/acceptance/node_test.go]: Going to try to instead install these globally in the docker image instead, so this shouldn't be a concern at all.

**Comment by couchand** [comment]
:lgtm: much nicer to work with, thanks for taking this on!

---

Reviewed 8 of 8 files at r1.
Review status: all files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---

*[pkg/acceptance/node_test.go, line 40 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19350#-KwkiaKpzSgaNQ3dH1Aj-r1-40:-Kwp-QlFO32OMdLKXzAq:b-oqttzu) ([raw file](https://github.com/cockroachdb/cockroach/blob/5c91c6cf9edb11c916e6387da5e798873e93ae31/pkg/acceptance/node_test.go#L40)):*
<details><summary><i>Previously, justinj (Justin Jaffray) wrote…</i></summary><blockquote>

Going to try to instead install these globally in the docker image instead, so this shouldn't be a concern at all.
</blockquote></details>

:+1: for adding the dependencies to the image.

---

*[pkg/acceptance/testdata/node/base-test.js, line 5 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19350#-Kwp2cb3J8Dld9_lKm8G:-Kwp2cb3J8Dld9_lKm8H:b-ctnotp) ([raw file](https://github.com/cockroachdb/cockroach/blob/5c91c6cf9edb11c916e6387da5e798873e93ae31/pkg/acceptance/testdata/node/base-test.js#L5)):*
> ```JavaScript
> 
> const errorsWith = require('./errors-with');
> ```

nit no space

---

*[pkg/acceptance/testdata/node/base-test.js, line 6 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19350#-Kwp2Z6wA_aCstjOeMke:-Kwp2Z6wA_aCstjOeMkf:bfniibl) ([raw file](https://github.com/cockroachdb/cockroach/blob/5c91c6cf9edb11c916e6387da5e798873e93ae31/pkg/acceptance/testdata/node/base-test.js#L6)):*
> ```JavaScript
> const errorsWith = require('./errors-with');
> 
> let client = require('./client');
> ```

s/let/const/

---

*[pkg/acceptance/testdata/node/base-test.js, line 10 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19350#-Kwp2kpIIR54eA2wmxXX:-Kwp2kpIIR54eA2wmxXY:bnz7t1w) ([raw file](https://github.com/cockroachdb/cockroach/blob/5c91c6cf9edb11c916e6387da5e798873e93ae31/pkg/acceptance/testdata/node/base-test.js#L10)):*
> ```JavaScript
> // We orchestrate a failure here to ensure that a failing test actually causes
> // the docker build to fail.
> describe('failure smoke test', () => {
> ```

if it were me I'd wrap this whole thing in `if(process.env.SHOULD_FAIL)` so it doesn't show up otherwise, then just `assert(false)`

---

*[pkg/acceptance/testdata/node/client.js, line 2 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19350#-Kwp2SO2IPkTHacKwih4:-Kwp2SO2IPkTHacKwih5:b-pye2iz) ([raw file](https://github.com/cockroachdb/cockroach/blob/5c91c6cf9edb11c916e6387da5e798873e93ae31/pkg/acceptance/testdata/node/client.js#L2)):*
> ```JavaScript
> const pg = require('pg');
> var fs = require('fs');
> ```

s/var/const/

---

*[pkg/acceptance/testdata/node/client.js, line 8 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19350#-Kwp1ZULMLqSmCdjRvjr:-Kwp1ZULMLqSmCdjRvjs:bok8qz) ([raw file](https://github.com/cockroachdb/cockroach/blob/5c91c6cf9edb11c916e6387da5e798873e93ae31/pkg/acceptance/testdata/node/client.js#L8)):*
> ```JavaScript
> // Cockroach instance.
> 
> let config;
> ```

it might be nice to always use `PGHOST` and `PGPORT` if available.  maybe something like:

```
const config = {
  user: 'root',
  host: process.env.PGHOST || 'localhost',
  port: process.env.PGPORT || 26257,
};
if (process.env.PGSSLCERT && process.env.PGSSLKEY) {
  config.ssl = {
      cert: fs.readFileSync(process.env.PGSSLCERT),
      key:  fs.readFileSync(process.env.PGSSLKEY),
  };
}
```

---

*[pkg/acceptance/testdata/node/client.js, line 27 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19350#-Kwp26kOSIFxCaiAjuu_:-Kwp26kPZKSFp3Xay4bX:bfniibl) ([raw file](https://github.com/cockroachdb/cockroach/blob/5c91c6cf9edb11c916e6387da5e798873e93ae31/pkg/acceptance/testdata/node/client.js#L27)):*
> ```JavaScript
> }
> 
> let client = new pg.Client(config);
> ```

s/let/const/

---

*[pkg/acceptance/testdata/node/client.js, line 28 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19350#-Kwp2MygZGezUiOj8t0m:-Kwp2MyhBMY45okoc2hE:bik9wdd) ([raw file](https://github.com/cockroachdb/cockroach/blob/5c91c6cf9edb11c916e6387da5e798873e93ae31/pkg/acceptance/testdata/node/client.js#L28)):*
> ```JavaScript
> 
> let client = new pg.Client(config);
> client.connect();
> ```

maybe move this into the before?

---

*[pkg/acceptance/testdata/node/errors-with.js, line 11 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19350#-Kwp0kD2tN5kY27KPyjT:-Kwp0kD2tN5kY27KPyjU:bgj9d51) ([raw file](https://github.com/cockroachdb/cockroach/blob/5c91c6cf9edb11c916e6387da5e798873e93ae31/pkg/acceptance/testdata/node/errors-with.js#L11)):*
> ```JavaScript
> // fails with a superstring of an asserted error message and a specified
> // Postgres error code.
> function errorsWith(promise, { msg, code }) {
> ```

nit: I'd name this `rejectsWith`

---

*[pkg/acceptance/testdata/node/errors-with.js, line 14 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19350#-Kwp-rFz_zAz8JChSpQO:-Kwp-rG--XyTPEtmP31R:bk8tlo8) ([raw file](https://github.com/cockroachdb/cockroach/blob/5c91c6cf9edb11c916e6387da5e798873e93ae31/pkg/acceptance/testdata/node/errors-with.js#L14)):*
> ```JavaScript
>   return promise.then(
>     () => {
>       assert.fail(`expected failure with message "${msg}" and Postgres error code ${pgCode}, but no failure occurred`);
> ```

s/pgCode/code/

---

*[pkg/acceptance/testdata/node/README.md, line 11 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19350#-Kwp1-is6XE-LQBqhSWE:-Kwp1-is6XE-LQBqhSWF:bgolypu) ([raw file](https://github.com/cockroachdb/cockroach/blob/5c91c6cf9edb11c916e6387da5e798873e93ae31/pkg/acceptance/testdata/node/README.md#L11)):*
> ```Markdown
> 
> * Run a Cockroach instance on the default port
> * `npm install && npm test`
> ```

Kind of odd to use `npm` here, when our usual JS package manager is `yarn`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/19350#-:-Kwp3cZzuXgLzwsxPz4U:bknz6ip)*

**Comment by justinj** [comment]
TFTRs!

**Pull request merged**